spec/01-unit/08-router_spec.lua

```diff
@@ -1669,6 +1669,198 @@ for _, flavor in ipairs({ "traditional", "traditional_compatible", "expressions"
         end)
       end)
 
+      if flavor ~= "traditional" then
+        describe("incremental rebuild", function()
+          local router
+          local use_case = {
+            {
+              service = service,
+              route = {
+                id = "e8fb37f1-102d-461e-9c51-6608a6bb8101",
+                paths = { "/foo", },
+                updated_at = 100,
+              },
+            },
+            {
+              service = service,
+              route = {
+                id = "e8fb37f1-102d-461e-9c51-6608a6bb8102",
+                paths = { "/bar", },
+                updated_at = 90,
+              },
+            }
+          }
+
+          before_each(function()
+            router = assert(new_router(use_case))
+          end)
+
+          it("matches initially", function()
+            local match_t = router:select("GET", "/foo")
+            assert.truthy(match_t)
+            assert.same(use_case[1].route, match_t.route)
+
+            match_t = router:select("GET", "/bar")
+            assert.truthy(match_t)
+            assert.same(use_case[2].route, match_t.route)
+          end)
+
+          it("update/remove works", function()
+            local use_case = {
+              {
+                service = service,
+                route = {
+                  id = "e8fb37f1-102d-461e-9c51-6608a6bb8101",
+                  paths = { "/foo1", },
+                  updated_at = 100,
+                },
+              },
+            }
+
+            local nrouter = assert(new_router(use_case, router))
+
+            assert.equal(nrouter, router)
+
+            local match_t = nrouter:select("GET", "/foo1")
+            assert.truthy(match_t)
+            assert.same(use_case[1].route, match_t.route)
+
+            match_t = nrouter:select("GET", "/bar")
+            assert.falsy(match_t)
+          end)
+
+          it("update skips routes if updated_at is unchanged", function()
```

Comment:
This case may not prove the 'skip unchanged' feature.

Now the check condition is `route_updated_at >= updated_at`, so the route will still be updated.

Perhaps in the future we can implement this.